internal/vm/checker/checker.go

```diff
@@ -27,16 +25,18 @@ type checker struct {
 
 	errors errors.ErrorList
 
-	depth   int
-	tooDeep bool
+	depth             int
+	tooDeep           bool
+	maxRecursionDepth int
+	maxRegexLength    int
 }
 
 // Check performs a semantic check of the astNode, and returns a potentially
 // modified astNode and either a list of errors found, or nil if the program is
 // semantically valid.  At the completion of Check, the symbol table and type
 // annotation are also complete.
-func Check(node ast.Node) (ast.Node, error) {
-	c := &checker{}
+func Check(node ast.Node, maxRegexLength int, maxRecursionDepth int) (ast.Node, error) { // TODO is there a way to inject these integers without adding them as part of this constructor?
+	c := &checker{maxRegexLength: maxRegexLength, maxRecursionDepth: maxRecursionDepth}
```

Comment:
Prior to this line, set the defaults.    This means you won't have to pass them from the flags in the test consturctors.

if maxRegexpLength == 0 { maxRegexpLength = kDefaultMaxregexpLenght }
if maxrecursion... { }